examples/bin/start-druid-main

```diff
@@ -0,0 +1,436 @@
+                        help='Validate the arguments and display memory distribution')
+    parser.add_argument('--run_zk',  action='store_true',
+                        help='Specification to run zookeeper, \n'
+                             'zk config is picked up from conf/zk.')
+    parser.add_argument('--verbose', action='store_true', help='Log details')
+
+    parser.set_defaults(run_zk=False)
+    parser.set_defaults(compute_only=False)
+    parser.set_defaults(verbose=False)
+
+    return parser
+
+def parse_arguments(args):
+    service_list = []
+    config_path = ""
+    total_memory = ""
+    compute_only = False
+    run_zk = False
+
+    if args.compute_only:
+        compute_only = True
+    if args.run_zk:
+        run_zk = True
+    if args.config_path is not None:
+        config_path = pathlib.Path(os.path.join(os.getcwd(), args.config_path)).resolve()
+        if os.path.exists(config_path) is False:
+            error_and_exit(f'config_path {config_path} doesn\'t exist')
+    if args.memory is not None:
+        total_memory = args.memory
+    if args.service_list is not None:
+        services = args.service_list.split(SERVICE_SEPARATOR)
+
+        for service in services:
+            if service not in DEFAULT_SERVICES:
+                error_and_exit(f'Invalid service name {service}, should be one of {DEFAULT_SERVICES}')
+
+            if service in service_list:
+                error_and_exit(f'{service} is specified multiple times')
+
+            service_list.append(service)
+
+    if len(service_list) == 0:
+        # start all services
+        service_list = DEFAULT_SERVICES
+        run_zk = True
+
+    return config_path, total_memory, service_list, run_zk, compute_only
+
+def print_startup_config(service_list, config_path, run_zk):
+    custom_print(f'starting {service_list}')
+    custom_print(f'reading config from {config_path}')
+    if run_zk:
+        zk_config_path = pathlib.Path(f'{os.getcwd()}/../conf/zk').resolve()
+        custom_print(f'starting zk, reading default config from {zk_config_path}')
+    custom_print('\n')
+
+def should_compute_memory(config_path, total_memory, service_list):
+    # if jvm file is present for any of the services
+    # it should be present for all services and memory should not be specified
+    # if memory is given, jvm file shouldn't be present for any service
+    jvm_config_count = 0
+    for service in service_list:
+        if pathlib.Path(f'{config_path}/{service}/jvm.config').is_file():
+            jvm_config_count += 1
+        elif jvm_config_count > 0:
+            error_and_exit(f'jvm.config file is missing for service {service}, jvm.config should be specified for all the services or none')
+
+    if jvm_config_count > 0 and (jvm_config_count != len(service_list) or total_memory != ""):
+        if jvm_config_count != len(service_list):
+            error_and_exit("jvm.config file should be present for all services or none")
```

Comment:
I think it would be better to raise an exception here (and in other such cases) rather than simply exiting. You can catch these exceptions where you are already catching the KeyboardInterrupt.

Reply:
Done. This looks better.